### PR TITLE
Update cookiecutter auth to DOT from DOP

### PR DESCRIPTION
**Pull request by tuchfarber**
Change the IDA cookiecutter so that OAuth2 authentication is used by default (and django-oauth-toolkit) instead of OpenID Connect (and django-oauth2-provider).

### CODE REVIEW

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/base.py [hunk 1]
```diff
@@ -164,12 +164,26 @@
...
+    'JWT_ALGORITHM': 'HS256',
+    'JWT_VERIFY_EXPIRATION': True,
+    'JWT_PAYLOAD_GET_USERNAME_HANDLER': lambda d: d.get('preferred_username'),
+    'JWT_LEEWAY': 1,
+    'JWT_DECODE_HANDLER': 'edx_rest_framework_extensions.auth.jwt.decoder.jwt_decode_handler',
+    'JWT_PUBLIC_SIGNING_JWK_SET': None,
+    'JWT_AUTH_COOKIE_HEADER_PAYLOAD': 'edx-jwt-cookie-header-payload',
+    'JWT_AUTH_COOKIE_SIGNATURE': 'edx-jwt-cookie-signature',
+    'JWT_AUTH_REFRESH_COOKIE': 'edx-jwt-refresh-cookie',
+}
```
Comment (nasthagiri) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/base.py:186]: Curious: from where did you find these settings?  Not sure if all of them are required.

File: {{cookiecutter.repo_name}}/docs/getting_started.rst [hunk 2]
```diff
@@ -22,51 +22,38 @@ When developing locally, it may be useful to have settings overrides that you do
...
+| Setting                           | Description                      | Value                                                                    |
++===================================+==================================+==========================================================================+
+| SOCIAL_AUTH_EDX_OAUTH2_KEY        | OAuth 2.0 client key             | (This should be set to the value generated when the client was created.) |
++-----------------------------------+----------------------------------+--------------------------------------------------------------------------+
+| SOCIAL_AUTH_EDX_OAUTH2_SECRET     | OAuth 2.0 client secret          | (This should be set to the value generated when the client was created.) |
++-----------------------------------+----------------------------------+--------------------------------------------------------------------------+
+| SOCIAL_AUTH_EDX_OAUTH2_URL_ROOT   | OAuth 2.0 authentication URL     | http://127.0.0.1:18000/oauth2                                            |
++-----------------------------------+----------------------------------+--------------------------------------------------------------------------+
```
Comment (nasthagiri) [inline {{cookiecutter.repo_name}}/docs/getting_started.rst:56]: Shall we add the Backend-specific  settings as well?  `BACKEND_SERVICE_EDX_OAUTH2_KEY` and `BACKEND_SERVICE_EDX_OAUTH2_SECRET`?

**Review by nasthagiri — APPROVED** [review]
Looks good overall!

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/base.py (hunk 1 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/base.py:186]: I got them from the Credentials IDA settings here:
https://github.com/edx/credentials/blob/master/credentials/settings/base.py#L361-L379
Happy to remove them if they aren't needed - are they needed?

File: {{cookiecutter.repo_name}}/docs/getting_started.rst (hunk 2 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/docs/getting_started.rst:56]: Yes - definitely! I will add them...

Comment (doctoryes) [reply {{cookiecutter.repo_name}}/docs/getting_started.rst:56]: Added.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/production.py [hunk 3]
```diff
@@ -12,11 +12,28 @@
...
 CONFIG_FILE = get_env_setting('{{cookiecutter.repo_name|upper}}_CFG')
 with open(CONFIG_FILE, encoding='utf-8') as f:
-    config_from_yaml = yaml.load(f)
+    config_from_yaml = yaml.safe_load(f)
+
+    # Remove the items that should be used to update dicts, and apply them separately rather
+    # than pumping them into the local vars.
+    dict_updates = {key: config_from_yaml.pop(key, None) for key in DICT_UPDATE_KEYS}
+
+    for key, value in list(dict_updates.items()):
```
Comment (pwnage101) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/production.py]: Nit: what's with list()ing an iterator?  Couldn't this just be `for key, value in dict_updates.items():`?

**Review by pwnage101 — APPROVED** [review]

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/production.py (hunk 3 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/settings/production.py]: Yes! Changed it.

**Comment by doctoryes** [comment]
Thanks for starting on this, @tuchfarber ! Now merged....

**Comment by tuchfarber** [comment]
Woah, thanks for doing this! I started it when building Designer, and totally forgot about finishing it up.

**Pull request merged**